AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -28,22 +28,30 @@ @implementation ASCollectionDataController {
   NSMutableDictionary *_pendingIndexPaths;
 }
 
-- (void)prepareForReloadData
+- (instancetype)initWithAsyncDataFetching:(BOOL)asyncDataFetchingEnabled
 {
-  _pendingNodes = [NSMutableDictionary dictionary];
-  _pendingIndexPaths = [NSMutableDictionary dictionary];
+  self = [super initWithAsyncDataFetching:asyncDataFetchingEnabled];
+  if (self != nil) {
+    _pendingNodes = [NSMutableDictionary dictionary];
+    _pendingIndexPaths = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
 
-  [[self supplementaryKinds] enumerateObjectsUsingBlock:^(NSString *kind, NSUInteger idx, BOOL *stop) {
+- (void)prepareForReloadData
+{
+  NSArray *kinds = [self supplementaryKinds];
+  for (NSString *kind in kinds) {
```

Comment:
Thanks for using fast enumeration!  No need for block overhead.  In this case I think it would be OK (although not much better) to call [self supplementaryKinds] in the for (), as it will only be called once.